server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -1195,14 +1196,25 @@ static IndexMetadata buildIndexMetadata(
         return blocksBuilder;
     }
 
-    private static void updateIndexMappingsAndBuildSortOrder(
+    private void updateIndexMappingsAndBuildSortOrder(
         IndexService indexService,
         CreateIndexClusterStateUpdateRequest request,
         List<Map<String, Object>> mappings,
         @Nullable IndexMetadata sourceMetadata
     ) throws IOException {
         MapperService mapperService = indexService.mapperService();
-        for (Map<String, Object> mapping : mappings) {
+        List<Map<String, Object>> mergedMappings = new ArrayList<>();
+        boolean isTimeSeriesMode = indexService.getIndexSettings().getMode() != null
+            && indexService.getIndexSettings().getMode() == IndexMode.TIME_SERIES;
```

Comment:
I think `IndexSettings.getMode` can't be null - it'd default to STANDARD.

I think I'd prefer to move this merge helping code to `IndexMode` too, if we can. If every behavior change in different index modes is a method on `IndexMode` then folks reading it can scan it for the kinds of behavior that it influences. Hard checks for a particular mode like this make that much harder.

Reply:
fine, I'm trying to move these code into IndexMode

Reply:
I change the code, is it ok?